components/grade-table.js

```diff
@@ -11,21 +12,65 @@ class GradeTable{
         for(var i=0; i<grades.length; i++){
             var gradesIndex = grades[i];
 
-            var tableRow = document.createElement('tr');
+            this.renderGradeRow(gradesIndex,this.deleteGrade);
 
-            var studentName = document.createElement('td');
-            studentName.textContent = gradesIndex.name;
-            tableRow.appendChild(studentName);
+            if(grades.length === 0){
+                var pEl = document.getElementById('no-grades');
+                pEl.classList.remove("d-none");
+            } 
 
-            var studentCourse = document.createElement('td');
-            studentCourse.textContent = gradesIndex.course;
-            tableRow.appendChild(studentCourse);
+            // tableBodyEl.appendChild(renderRow);
 
-            var studentGrade = document.createElement('td');
-            studentGrade.textContent = gradesIndex.grade;
-            tableRow.appendChild(studentGrade);
+            // var tableRow = document.createElement('tr');
 
-            tableBodyEl.appendChild(tableRow);
+            // var studentName = document.createElement('td');
+            // studentName.textContent = gradesIndex.name;
+            // tableRow.appendChild(studentName);
+
+            // var studentCourse = document.createElement('td');
+            // studentCourse.textContent = gradesIndex.course;
+            // tableRow.appendChild(studentCourse);
+
+            // var studentGrade = document.createElement('td');
+            // studentGrade.textContent = gradesIndex.grade;
+            // tableRow.appendChild(studentGrade);
+
+            // tableBodyEl.appendChild(tableRow);
         }
     }
+    onDeleteClick(deleteGrade){
+        this.deleteGrade = deleteGrade;
+
+    }
+    renderGradeRow(data,deleteGrade){
+        var tableBodyEl = this.tableElement.querySelector('tbody');
+
+        var tableRow = document.createElement('tr');
+
+        var studentName = document.createElement('td');
+        studentName.textContent = data.name;
+        tableRow.appendChild(studentName);
+
+        var studentCourse = document.createElement('td');
+        studentCourse.textContent = data.course;
+        tableRow.appendChild(studentCourse);
+
+        var studentGrade = document.createElement('td');
+        studentGrade.textContent = data.grade;
+        tableRow.appendChild(studentGrade);
+
+        var operationsCol = document.createElement('td');
+        tableRow.appendChild(operationsCol);
+
+        var createButton = document.createElement('button');
+        createButton.classList = 'btn btn-danger';
+        createButton.textContent = 'Delete';
+        operationsCol.appendChild(createButton);
+        // createButton.addEventListener('click', deleteGrade(data.id));
+        createButton.addEventListener('click', function(){
+            deleteGrade(data.id);
+        });
+
+        tableBodyEl.appendChild(tableRow);
+    }
```

Comment:
As stated above, this method should create a table row (and its contents) and return it, not append it to the table body.